jmh/benchmark-binary-tree/README.md

```diff
@@ -7,6 +7,17 @@ configuration can be found in _src/main/resources/META-INF/native-image/_.
 
 > **Note:** To generate the configuration yourself, ensure that the JMH `fork` parameter is set to `0`, which can be performed from the command line using the option  `-f 0`. It can also be achieved within the code by using the `@Fork` annotation.
 
+## Important Notes on Using JMH with GraalVM Native Image
+
+In order to make a JMH benchmark run as native executable, built by GraalVM Native Image, you can not fork the benchmark process 
+in the same way as JMH does when running on the JVM. When running on the JVM JMH will fork a new JVM for each benchmark in order
+to ensure there is no interference in the measurements for each benchmark. This forking process is not possible on GraalVM Native Image
+and you should consider the following guidance when building JMH benchmarks that are meant to be run as native executables:
+
+* You should only include a single benchmark in each native executable
+* You need to annotate the benchmark with, `@Fork(0)` to ensure that the benchmark is not forked
+* If you want to profile the benchmark in order to generate an optimised benchmark, you should, obviously, ignore the benchmark results whilst profiling
```

Comment:
... in order to generate an optimised benchmark > ... to generate an optimized benchmark 
(don't use "in order", not recommended nowadays)